### ISSUE TITLE
Add test for $search

### ISSUE THREAD

**Issue by ekryski** [feathersjs/feathers-service-tests#7]
Once this is implemented we should bump the major or minor version so that the builds for database adapters that don't have this implemented yet don't just fail.

What we probably need to test:
- mixed case complete matching
- where search makes up substring

The suggested syntax is:

``` js
name: {
  $search: ['alice', 'Alice', 'bo', /$bob/i]
}
```

**Comment by daffl** [comment]
I'm wondering if instead of having a separate issue for this in each db adapter we should just track one linked with all the discussions around it in https://github.com/feathersjs/feathers much like https://github.com/feathersjs/feathers/issues/219 (which does require to change all adapters, too).

**Comment by ekryski** [comment]
I agree that we should have a central place to track progress but I think we should have issues in each repo. It makes it easier for people to see which is done and for us to just close as part of a PR or commit.

Currently, I've already added them to all the repos the main release issue so people have better visibility.

**Comment by daffl** [comment]
I started on implementing this and took the liberty to bulldoze through and close all the related issues 😉

**Comment by ekryski** [comment]
Once this lands we'll probably need to bump the version of this repo so that adapters that don't have this functionality don't look like they are failing.

**Comment by daffl** [comment]
Closing. See discussion in https://github.com/feathersjs/feathers/issues/334#issuecomment-260834107
